process/smartContract/hooks/vmAccountsDB.go

```diff
@@ -141,9 +184,134 @@ func (vadb *VMAccountsDB) GetCode(address []byte) ([]byte, error) {
 	return code, nil
 }
 
-// GetBlockhash is deprecated
+// GetBlockhash returns the header hash for a requested nonce delta
 func (vadb *VMAccountsDB) GetBlockhash(offset *big.Int) ([]byte, error) {
-	return nil, nil
+	if offset.Cmp(big.NewInt(0)) > 0 {
+		return nil, process.ErrInvalidNonceRequest
+	}
+
+	hdr := vadb.blockChain.GetCurrentBlockHeader()
+
+	requestedNonce := big.NewInt(0).SetUint64(hdr.GetNonce())
+	requestedNonce.Sub(requestedNonce, offset)
+
+	if requestedNonce.Cmp(big.NewInt(0)) < 0 {
+		return nil, process.ErrInvalidNonceRequest
+	}
+
+	if offset.Cmp(big.NewInt(0)) == 0 {
+		return vadb.blockChain.GetCurrentBlockHeaderHash(), nil
+	}
+
+	if vadb.shardCoordinator.SelfId() == sharding.MetachainShardId {
```

Comment:
the process inside is of type MetaProcessor or ShardProcessor, could we simplify this by calling:

GetHeaderFromStorage(
requestedNonce.Uint64(),
		vadb.shardCoordinator.SelfId(),
		vadb.storageService,
		vadb.uint64Converter,
		vadb.marshalizer,
)

And behind call the different methods?